src/common/mkstemp.cpp

```diff
@@ -0,0 +1,108 @@
+/* mkstemp extracted from libc/sysdeps/posix/tempname.c.  Copyright
+   (C) 1991-1999, 2000, 2001, 2006 Free Software Foundation, Inc.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
```

Comment:
Is it compatible with RNP license to include pieces of LGPL-licensed code? ping @ronaldtse